azerothcore/azerothcore-wotlk
# fix(CORE/Commands): .cheat god lowering hp to 1 and .cheat power not refilling power

## CHANGES PROPOSED:
-  On power resource update to have maximum value if .cheat power is enabled
-  Dont check if player has enough power to cast spell if .cheat power is enabled
-  Changed SendSpellNonMelee...Log to send damage as absorb instead of damage to avoid client prediction lowering hp bar when .cheat god is enabled but still be able to debug damage from enemies (looking at absorb log)
- Avoid fall damage while .god mode is enabled


## ISSUES ADDRESSED:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3257
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1534

## TESTS PERFORMED:
- Activated god mode (.cheat god) and infinite power (.cheat power) on a warrior, dk, rouge and mage
- Went to ICC Raid
- Pulled all till first boss and all attacking for 5 minutes
- Checked my hp bar is 100%
- Attacked mobs with many different spells that consume power
- Checked that power bar ddi not get below 95%



## HOW TO TEST THE CHANGES:

For .cheat god
1. Activate god mode (.cheat god)
2. Go to ICC or any other instance zone with many mobs
3. Pull many mobs and let them attack for  a while
4. Check that HP bar did not go down

For .cheat power
1. Create class (warrior, DK, rouge and any Mana clasS)
2. Activate .cheat power on each
3. Use spells
4. Check that resource did not go down and its always at 95%-100% even outside combat.

## KNOWN ISSUES AND TODO LIST:



## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
